05_testing_and_ci/automation_exercise.md

```diff
@@ -0,0 +1,92 @@
+- Add the contents of the [automation exercise repository (TODO: Update link)](https://github.com/Simulation-Software-Engineering/automation-exercise-preparation) to your own repository. You can do this by copying the files into your own repository, adding and committing them to Git. Afterwards push the changes to GitHub and verify that your repository contains the same files as the `automation-exercise` repository.
+
+  **Note**: If you use another way of adding the data to your repository, e.g., by defining a new remote, make sure that the `origin` remote points to your own repository.
+
+- Set up the GitHub Action to run the checks, tests, and to create the coverage report.
+    - The action should have three jobs (`style_check`, `test`, `coverage_report`). Each of these jobs should carry out the corresponding step described in [Task descriptions](## Task descriptions).
+    - The action should be triggered by `push` events.
+    - The intermediate coverage information (`.coverage`) should be kept for one day. This is the minimum amount of time the artifact must be stored.
+    - Make sure that in each job all of the needed Python dependencies are installed (for testing and for running the code) within the action. If you forgot, what the dependencies were, check out the [Python testing exercise sheet](https://github.com/Simulation-Software-Engineering/Lecture-Material/blob/main/05_testing_and_ci/python_testing_exercise.md).
+
+- Add a [GitHub workflow status badge](https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/adding-a-workflow-status-badge) to the README that is based on your workflow and the main branch.
+    - Label this badge with "SSE CI/CD".
+
+- Submit your solution via an issue. You cannot submit a pull request since you cannot use a fork in the exercise. You can also add the issue before you are done in order to ask questions. If you are opening an issue before you are finished, please add comment in the issue mentioning that it is "work in progress". When you are finished, please add another comment that the issue (or rather your repository) is "ready for review".
+    - Create an issue in the [automation exercise repository (TODO: Update link)](https://github.com/Simulation-Software-Engineering/automation-exercise-preparation).
+    - The issue should be named `[USERNAME] GitHub Actions exercise`, e.g., `[jaustar] GitHub Actions exercise`. Please use the GitLab username here.
+    - Please add a link to your GitHub repository.
+
+## GitLab CI/CD
+
+In this subtask you should implement the workflow using GitLab and add a GitLab Runner to your repository.
+
+- Create a **fork** of the GitLab repository [automation exercise repository (TODO: Update link)](https://gitlab-sim.informatik.uni-stuttgart.de/sse-test-group/automation-exercise).
```

Comment:
For adding runners one only needs the URL and the secret token. Besides that I am not always sure what is allowed/visible by default. Since I am admin in the GitLab instance I can see everything and also edit everything.

Reply:
I will ask the students to add us as members of the fork project such that we are guaranteed to see it.